corehq/motech/repeaters/models.py

```diff
@@ -250,6 +259,80 @@ def reset_next_attempt(self):
             self.save()
 
 
+class SQLCaseRepeater(SyncSQLToCouchMixin, SQLRepeater):
+    """
+    Record that cases should be repeated to a new url
+
+    """
+
+    payload_generator_classes = (CaseRepeaterXMLPayloadGenerator, CaseRepeaterJsonPayloadGenerator)
+
+    version = models.CharField(
+        max_length=10,
+        choices=list(zip(LEGAL_VERSIONS, LEGAL_VERSIONS)),
+        default=V2,
+    )
+    white_listed_case_types = ArrayField(
+        models.CharField(max_length=128, default=list)
+    )
+    black_listed_users = ArrayField(
+        models.CharField(max_length=128, default=list)
+    )
+    # users who caseblock submissions should be ignored
+    friendly_name = _("Forward Cases")
+
+    format = models.CharField(max_length=16)
```

Comment:
> Additionally, in the couch models this is defined on the Repeater class (and has no length limit). 

@calellowitz  There are only two possible values of the `format` property `case_xml` and `case_json`. So I kept the length to be 16.

> Why is it on the subclass not the base class here, and where does the length limit come from?

While looking into other repeaters, I found out that only FormRepeater, CaseRepeater use this property. So I thought of moving it to subclass rather than on base. For the rest of the Repeaters, it is set to `''` an empty string in DB.

> Adding to what Cal said, if you are wanting to overload Repeater.format, I would use the choices=... parameter to show where the possible values for CaseRepeater.format come from, and to show that they correspond to the payload generators.

Adding `choices` would be a good idea, thanks for the suggestion.

> It is important to note that you cannot override model fields if you are using multi-table inheritance. You can only do it if you are using abstract class inheritance, as Cal asked earlier.
If you are using multi-table inheritance (which is currently true) then the format field should be defined on the parent class and cannot be changed on child classes. ... (Or only defined on final child classes.)

Thanks for sharing the info @kaapstorm. Do you think it would be a good idea to have `format` property in the parent class even if it is being used only in a few child classes?

Reply:
I think the authors of the Repeater class thought `format` would be used more, and that's why it was defined on the base class. But it turns out we don't use it all that much. Personally, I'd move it to the child classes that use it.

Reply:
I know its only used twice right now and both are under 16, but 16 is pretty tight, and doesnt really provide any benefit  since it is already constrained to a set of choices. It is easily conceivable we will have a format longer than 16 in the future. Would you consider making this something like 32, just so we have the flexibilty, in case we end up needing to use more complex formats at some point in the future (like integrations with other types of systems that have their own unique format)